packages/cli/test/integration/credentials.ee.test.ts

```diff
@@ -0,0 +1,270 @@
+	const savedCredential = await saveCredential(credentialPayload(), { user: owner });
+
+	const response = await agent
+		.post(`/credentials/${savedCredential.id}/share`)
+		.send({ shareeId: 'abc' });
+
+	expect(response.statusCode).toBe(400);
+});
+
+test('POST /credentials/:id/share should fail if no sharee id is provided', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
+	const savedCredential = await saveCredential(credentialPayload(), { user: owner });
+
+	const response = await agent.post(`/credentials/${savedCredential.id}/share`).send();
+
+	expect(response.statusCode).toBe(400);
+});
+
+// ----------------------------------------
+// unshare
+// ----------------------------------------
+
+test('DELETE /credentials/:id/share should unshare the credential', async () => {
+	const ownerShell = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: ownerShell });
+	const savedCredential = await saveCredential(credentialPayload(), { user: ownerShell });
+
+	const response = await agent
+		.delete(`/credentials/${savedCredential.id}/share`)
+		.send({ shareeId: member.id });
+
+	expect(response.statusCode).toBe(200);
+	expect(response.body.data).toBeUndefined();
+
+	// check if the entry is removed from DB
+	const sharedCredential = await Db.collections.SharedCredentials.findOne({
+		where: { credentials: savedCredential, user: member },
+	});
+	expect(sharedCredential).toBeUndefined();
+
+	// check if credential in DB is untouched
+	const credential = await Db.collections.Credentials.findOneOrFail(savedCredential.id);
+	expect(credential).toEqual(savedCredential);
+});
+
+test('DELETE /credentials/:id/share should fail for non-existing credentials', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
+
+	const response = await agent.delete(`/credentials/1234567/share`).send({ shareeId: member.id });
+
+	expect(response.statusCode).toBe(403);
+});
+
+test('DELETE /credentials/:id/share should be forbidden for non-owned credentials', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
+	const savedCredential = await saveCredential(credentialPayload(), { user: member });
+
+	const response = await agent
+		.delete(`/credentials/${savedCredential.id}/share`)
+		.send({ shareeId: member.id });
+
+	expect(response.statusCode).toBe(403);
+});
+
+test('DELETE /credentials/:id/share should be indempotent', async () => {
```

Comment:
with `404` you would communicate that the request has failed because the resource was not found.

In this case we don't care about this, we just want to ensure that the resource is removed. Accordingly if the resource did not exist we achieved our goal that the resource is removed, thus a `200`